doc/languages-frameworks/go.xml

```diff
@@ -3,12 +3,107 @@
          xml:id="sec-language-go">
  <title>Go</title>
 
- <para>
-  The function <varname>buildGoPackage</varname> builds standard Go programs.
- </para>
+ <section xml:id="ssec-go-modules">
+  <title>Go modules</title>
 
- <example xml:id='ex-buildGoPackage'>
-  <title>buildGoPackage</title>
+  <para>
+   The function <varname> buildGoModule </varname> builds Go programs managed
+   with Go modules. It builds a
+   <link xlink:href="https://github.com/golang/go/wiki/Modules">Go
+   modules</link> through a two phase build:
+   <itemizedlist>
+    <listitem>
+     <para>
+      An intermediate fetcher derivation. This derivation will be used to fetch
+      all of the dependencies of the Go module.
+     </para>
+    </listitem>
+    <listitem>
+     <para>
+      A final derivation will use the output of the intermediate derivation to
+      build the binaries and produce the final output.
+     </para>
+    </listitem>
+   </itemizedlist>
+  </para>
+
+  <example xml:id='ex-buildGoModule'>
+   <title>buildGoModule</title>
+<programlisting>
+pet = buildGoModule rec {
```

Comment:
By default buildGoPackage was a split output, $bin contains the executables
and $out contained the source. Since we did not had a use-case, source from $out was dropped.
Since then $out is empty for many packages. I wonder if for `buildGoModule` we just not start using multiple outputs, since we don't really need them.

Reply:
I've updated the function and fix gotools that got broken by this change.